src/Microsoft.DotNet.Build.Tasks/GenerateTestExecutionScripts.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateTestExecutionScripts : Task
+    {
+        [Required]
+        public string[] TestCommands { get; set; }
+
+        [Required]
+        public ITaskItem[] TestDependencies { get; set; }
+
+        [Required]
+        public string RunnerScriptTemplate { get; set; }
+
+        [Required]
+        public string ScriptOutputPath { get; set; }
+
+        public override bool Execute()
+        {
+            if (!File.Exists(RunnerScriptTemplate))
+            {
+                throw new FileNotFoundException($"Runner script template {RunnerScriptTemplate} was not found.");
+            }
+
+            string executionScriptTemplate = File.ReadAllText(RunnerScriptTemplate);
+            Directory.CreateDirectory(Path.GetDirectoryName(ScriptOutputPath));
+
+            Log.LogMessage($"Test Command lines = {string.Join(Environment.NewLine, TestCommands)}");
+            string extension = Path.GetExtension(Path.GetFileName(ScriptOutputPath)).ToLowerInvariant();
+            switch (extension)
+            {
+                case ".sh":
+                    WriteShExecutionScript(executionScriptTemplate, ScriptOutputPath);
+                    break;
+                case ".cmd":
+                case ".bat":
+                    WriteCmdExecutionScript(executionScriptTemplate, ScriptOutputPath);
+                    break;
+                default:
+                    throw new System.NotSupportedException($"Generating runner scripts with extension '{extension}' is not yet supported");
+            }
+            return true;
+        }
+
+        private void WriteShExecutionScript(string shExecutionTemplate, string outputPath)
+        {
+            // Build up the copy commands... 
+            StringBuilder copyCommands = new StringBuilder();
+            foreach (ITaskItem dependency in TestDependencies)
+            {
+                string relativePath = dependency.GetMetadata("PackageRelativePath");
+                bool? useAbsolutePath = dependency.GetMetadata("UseAbsolutePath")?.Equals("true", StringComparison.OrdinalIgnoreCase);
+                if (useAbsolutePath == true)
+                {
+                    copyCommands.Append($"cp -l -n \"{dependency.GetMetadata("SourcePath")}\" \"$EXECUTION_DIR/{Path.GetFileName(relativePath)}\" || exit $?\n");
```

Comment:
Should this be something different than `Path.GetFileName(relativePath)` since you haven't checked if `relativePath` is defined?